torch2jax/__init__.py

```diff
@@ -211,6 +218,35 @@ def _coerce(x):
     )
 
 
+def _tree_coerce(x):
+  spec = torch_tree_structure(x)
+  specs = deque([spec])
+
+  while len(specs) > 0:
+    spec = specs.popleft()
+    specs.extend(spec.children_specs)
+    if spec.is_leaf():
+      continue
+    if spec.type in jax._src.tree_util._registry:
+      continue  # already registered
+    node = torch.utils._pytree.SUPPORTED_NODES.get(spec.type, None)
+    if node is None:
+      continue
+    if is_dataclass(spec.type):
+      jax.tree_util.register_dataclass(spec.type)
+    else:
+      jax.tree_util.register_pytree_node(
+        spec.type,
+        node.flatten_fn,
+        lambda a, b, n=node: n.unflatten_fn(b, a),  # torch & jax have different order of args
```

Comment:
```suggestion
        lambda a, b: node.unflatten_fn(b, a),  # torch & jax have different order of args
```
docs suggest that this function takes exactly two arguments

Reply:
The trick here is to capture the node at the time of closure creation, otherwise the value of the node will change.

Reply:
i'm not sure i follow... the lambda will capture `node` in its closure by value not by name. what would be a test case that would make this fail?

Reply:
wow, that is... terrifying! TIL